doc/compatibility.rst

```diff
@@ -0,0 +1,97 @@
+Compatibility with GMT/Python/NumPy versions
```

Comment:
Should include this file in the `index.rst` file. Maybe put it under the "Reference Documentation" section, just under the "Changelog" entry?

Reply:
This is a good idea and is done in commit https://github.com/GenericMappingTools/pygmt/pull/2862/commits/84f02a916b229995b9a2ec4f3eb14578c8e44443.